### PR TITLE
LPS-157240

### PR DESCRIPTION
**Pull request by rodrigopaulino**
Related ticket: https://issues.liferay.com/browse/LPS-157240

### CODE REVIEW

**Comment by rodrigopaulino** [comment]
ci:test:sf

**Comment by rodrigopaulino** [comment]
ci:test:relevant

**Comment by rodrigopaulino** [comment]
ci:test:sf

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectRelationshipLocalServiceTest.java [hunk 1]
```diff
@@ -368,67 +444,217 @@ private void _testSystemObjectRelationshipManyToMany() throws Exception {
...
+		}
+		catch (ObjectRelationshipParameterObjectFieldIdException
+					objectRelationshipParameterObjectFieldIdException) {
+
+			Assert.assertEquals(
+				"Object relationship type " +
+					ObjectRelationshipConstants.TYPE_MANY_TO_MANY +
+						" does not allow a parameter object field ID",
+				objectRelationshipParameterObjectFieldIdException.getMessage());
+		}
```
Comment (carolmariaabb) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectRelationshipLocalServiceTest.java:466]: I think Brian usually sorts by exception name, so this would have to come before the `ObjectRelationshipTypeException`

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectRelationshipLocalServiceTest.java [hunk 2]
```diff
@@ -60,6 +74,49 @@ public class ObjectRelationshipLocalServiceTest {
...
+				null,
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+				"A" + RandomTestUtil.randomString(), null, null,
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+				ObjectDefinitionConstants.SCOPE_COMPANY, 1,
+				Arrays.asList(
+					ObjectFieldUtil.createObjectField(
+						"Text", "String", RandomTestUtil.randomString(),
+						StringUtil.randomId())));
+
```
Comment (carolmariaabb) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectRelationshipLocalServiceTest.java:91]: Can we move the creation of the `_systemObjectDefinition2` to this method?

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectRelationshipLocalServiceTest.java [hunk 3]
```diff
@@ -368,67 +444,217 @@ private void _testSystemObjectRelationshipManyToMany() throws Exception {
...
+			_systemObjectDefinition2.getObjectDefinitionId(), 0,
 			ObjectRelationshipConstants.DELETION_TYPE_PREVENT,
 			LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
 			StringUtil.randomId(),
 			ObjectRelationshipConstants.TYPE_MANY_TO_MANY);
 	}
 
 	private void _testSystemObjectRelationshipOneToMany() throws Exception {
-		ObjectRelationship objectRelationship =
+		ObjectRelationship objectRelationship1 =
```
Comment (carolmariaabb) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectRelationshipLocalServiceTest.java:479]: Why did you rename it to `objectRelationship1`? I couldn't find the `objectRelationship2`

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectRelationshipLocalServiceTest.java [hunk 4]
```diff
@@ -368,67 +444,217 @@ private void _testSystemObjectRelationshipManyToMany() throws Exception {
...
+				objectRelationshipParameterObjectFieldIdException.getMessage());
+		}
+
+		List<ObjectField> objectDefinition2ObjectFields =
+			_objectFieldLocalService.getObjectFields(
+				_objectDefinition2.getObjectDefinitionId());
+
+		ObjectField objectDefinition2ObjectField =
+			objectDefinition2ObjectFields.get(0);
+
```
Comment (carolmariaabb) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectRelationshipLocalServiceTest.java:559]: Not sure about these names. I think we could just call them  `objectFields` and `objectField`.

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectRelationshipLocalServiceTest.java (hunk 2 above)
Comment (rodrigopaulino) [reply modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectRelationshipLocalServiceTest.java:91]: The reason I made this choice is because I don't need it to be created before every test. This actually increases individual test time and I depend on this system object definition because I need to register a service there and I need its name.

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectRelationshipLocalServiceTest.java (hunk 3 above)
Comment (rodrigopaulino) [reply modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectRelationshipLocalServiceTest.java:479]: You are right. I think I used to have one, but forgot to rename it after I removed it.

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectRelationshipLocalServiceTest.java (hunk 4 above)
Comment (rodrigopaulino) [reply modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectRelationshipLocalServiceTest.java:559]: Sure thing. I'll make this change

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectRelationshipLocalServiceTest.java (hunk 1 above)
Comment (rodrigopaulino) [reply modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectRelationshipLocalServiceTest.java:466]: Didn't know that. I'll make this change.

Comment (rodrigopaulino) [reply modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectRelationshipLocalServiceTest.java]: Done

**Comment by rodrigopaulino** [comment]
ci:test:sf

**Comment by rodrigopaulino** [comment]
ci:test:relevant

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectRelationshipLocalServiceTest.java (hunk 1 above)
Comment (rodrigopaulino) [reply modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectRelationshipLocalServiceTest.java]: Done again

**Comment by rodrigopaulino** [comment]
ci:test:sf

**Comment by rodrigopaulino** [comment]
ci:test:relevant

**Review by carolmariaabb — APPROVED** [review]

**Comment by carolmariaabb** [comment]
ci:forward

**Comment by carolmariaabb** [comment]
ci:reevaluate:1377401_3227

**Comment by rodrigopaulino** [comment]
ci:close

**Comment by rodrigopaulino** [comment]
Resent https://github.com/liferay-forms/liferay-portal/pull/1842
